kyo-scheduler/jvm/src/main/scala/kyo/scheduler/Queue.scala

```diff
@@ -1,27 +1,28 @@
-package kyo.scheduler.util
+package kyo.scheduler
 
+import java.lang.invoke.VarHandle
 import java.util.concurrent.atomic.AtomicBoolean
 import kyo.*
 import scala.collection.mutable.PriorityQueue
 
-final private[kyo] class Queue[T](using ord: Ordering[T]) extends AtomicBoolean:
+final private class Queue[T](using ord: Ordering[T]) extends AtomicBoolean:
 
     private val queue = PriorityQueue[T]()
 
-    @volatile private var items = 0
+    private var items = 0
 
     def isEmpty() =
-        items == 0
+        size() == 0
 
     def size(): Int =
+        VarHandle.acquireFence()
         items
```

Comment:
Force a read barrier since the field isn't volatile anymore so all threads see the latest value.